android/src/main/java/com/lwansbrough/RCTCamera/RCTCameraModule.java

```diff
@@ -162,6 +183,185 @@ public String getName() {
         });
     }
 
+    private boolean prepareMediaRecorder(ReadableMap options) {
+        CamcorderProfile cm = RCTCamera.getInstance().setCaptureVideoQuality(options.getInt("type"), options.getString("quality"));
+
+        // Attach callback to handle maxDuration (@see onInfo method in this file)
+        mMediaRecorder.setOnInfoListener(this);
+        mMediaRecorder.setCamera(mCamera);
+        mCamera.unlock();  // make available for mediarecorder
+
+        // Set AV sources
+        mMediaRecorder.setAudioSource(MediaRecorder.AudioSource.CAMCORDER);
+        mMediaRecorder.setVideoSource(MediaRecorder.VideoSource.CAMERA);
+
+        int mediaRecorderHintOrientation = (90 + ((720 - RCTCamera.getInstance().getActualDeviceOrientation() * 90))) % 360;
+
+        // adjust for differences in how devices display front facing http://www.theverge.com/2015/11/9/9696774/google-nexus-5x-upside-down-camera
+        if (RCT_CAMERA_TYPE_FRONT == options.getInt("type")) {
+            if ((RCT_CAMERA_ORIENTATION_PORTRAIT == RCTCamera.getInstance().getOrientation()) && (RCTCamera.getInstance().getAdjustedDeviceOrientation() == 90)) {
+                mediaRecorderHintOrientation += 180;
+            }
```

Comment:
I see the 2 way front camera mounting issue, but I don't understand why only "trying to fix the upside down issue" when in portrait ? I tested on Nexus 5. The front camera recording is upside down but if I go landscape, the .mp4 recording is in the correct orientation.

Great work BTW. Thank you for adding this !